### ISSUE TITLE
Fix deprecation warnings in Gradle 4.8

### ISSUE THREAD

**Issue by atorok** [elastic/elasticsearch#31264]
The upgrade to Gradle 4.8 from #31230 come with some new deprecation warnings: 

`Creating a custom task named 'wrapper' has been deprecated and is scheduled to be removed in Gradle 5.0. You can configure the existing task using the 'wrapper { }' syntax or create your custom task under a different name.'.`

`As part of making the publishing plugins stable, the 'deferred configurable' behavior of the 'publishing {}' block is now deprecated. Please add 'enableFeaturePreview('STABLE_PUBLISHING')' to your settings file and do a test run by publishing to a local repository. If all artifacts are published as expected, there is nothing else to do. If the published artifacts change unexpectedly, please see the migration guide for more details: https://docs.gradle.org/4.8/userguide/publishing_maven.html#publishing_maven:deferred_configuration. In Gradle 5.0 the flag will be removed and the new behavior will become the default.`

```
The AbstractFileCollection.getBuildDependencies() method has been deprecated and is scheduled to be removed in Gradle 5.0. Do not extend AbstractFileCollection, use Project.files() instead.
	at org.gradle.api.internal.file.AbstractFileCollection.getBuildDependencies(AbstractFileCollection.java:209)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:93)
	at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmpyEdges.getNodeValues(CachingDirectedGraphWalker.java:211)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.doGetDependencies(CachingTaskDependencyResolveContext.java:76)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:60)
```

**Comment by jasontedor** [comment]
Relates #31230

**Comment by atorok** [comment]
the last one around `AbstractFileCollection` seems to be gralde relatet, but as gradle/gradle#5141 explains it has to do with johnrengelman/shadow#370. It's fixed by moving to version `2.0.4` of the plugin.
